fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -2429,14 +2492,37 @@ public Builder setMutualIndexingBoundaries(final List<Tuple> primaryKeysBoundari
                 return this;
             }
 
+            /**
+             * If the index is partly built and blocked, allowed (or disallow) unblocking before indexing.
+             * @param allowUnblock if true, unblock (if blocked) and continue.
+             * @param allowUnblockId if non-null and non-empty, unblock only if the indexing stamp's id matches it. Will not continue if not blocked.
```

Comment:
What happens if `allowUnblock = false` and `allowUnblockId` is not `null`?

It's also a little weird to me that if `allowUnblockId` is not `null`, then it won't build on an unblocked index build, but if `allowUnblockId` is `null`, it will. But on the other hand, that doesn't necessarily seem to be what `shouldAllowUnblock` is doing

Reply:
> what happens if
Unblocking is not allowed, so blockId is meaningless. 
Yet, the API had changed and the users cannot call this combination.